source/common/http/conn_manager_impl.cc

```diff
@@ -251,7 +251,8 @@ void ConnectionManagerImpl::doEndStream(ActiveStream& stream) {
   bool connection_close = stream.state_.saw_connection_close_;
   bool request_complete = stream.filter_manager_.remoteDecodeComplete();
 
-  checkForDeferredClose(connection_close && (request_complete || http_10_sans_cl));
+  checkForDeferredClose((connection_close && (request_complete || http_10_sans_cl)) ||
+                        stream.state_.is_tunneling_);
```

Comment:
Can you add some comments here on the various conditions? It's kind of complicated now and hard to quickly parse.

Reply:
OK I think I'm roughly following you here. I could go either way, but I would request:
1) More comments here for those that aren't as well versed in all things CONNECT
2) Release note
3) It seems like this should probably be guarded, but I will defer to your opinion on how risky this is.